### PR TITLE
Update mergo to version 0.3.2

### PR DESCRIPTION
**Pull request by ghost**
Fixes issue #27543. This is important for the Kubernetes packaging on Debian, see https://bugs.debian.org/878254 for a description of the situation there. For upstream Kubernetes very little (if anything) should change.

### CODE REVIEW

**Comment by vainu-arto** [comment]
/release-note-none

I assume this is correct, since there should not be any user visible changes.

**Comment by sttts** [comment]
/assign @ncdc 

This area is pretty fragile last time I looked at it.

**Comment by sttts** [comment]
> For upstream Kubernetes very little (if anything) should change.

It's not that simple. Please be careful, compare https://github.com/kubernetes/kubernetes/pull/20628 and https://github.com/kubernetes/kubernetes/pull/20579#issuecomment-179676596. The update to 0.2.4 was toxic at that time and broke the logic. Now you also switch to `MergeWithOverwrite`. Maybe that helps to work around my concerns.

**Comment by vainu-arto** [comment]
Indeed, the meaning of Merge changed in mergo. That is the source of the bug in Debian (who use upstream mergo instead of the vendored one). MergeWithOverwrite should do what Merge used to do in previous versions (horrible api maintenance by upstream). 

Assuming that the unit tests are complete enough this seems to hold (I have also done some local testing which agrees as well).

File: staging/src/k8s.io/client-go/tools/clientcmd/client_config_test.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func TestOldMergoLib(t *testing.T) {
 	}
 	dst := T{X: "one"}
 	src := T{X: "two"}
-	mergo.Merge(&dst, &src)
+	mergo.MergeWithOverwrite(&dst, &src)
```
Comment (sttts) [inline staging/src/k8s.io/client-go/tools/clientcmd/client_config_test.go]: Please update the test name and the comments below.

**Comment by sttts** [comment]
> who use upstream mergo instead of the vendored one

Why? Sounds pretty dangerous to replace arbitrary libs.

**Comment by sttts** [comment]
Please squash the last two commits.

**Comment by vainu-arto** [comment]
/retest

These now failing tests apparently did succeed before, and the code has not changed. I'm unsure why they were re-run.

**Comment by vainu-arto** [comment]
Huh? Are these tests actually failing since the year changed?

**Comment by sttts** [comment]
> Huh? Are these tests actually failing since the year changed?

There is a PR updating all the generated files to use 2018. Yes, our CI fails on that right now.

**Comment by vainu-arto** [comment]
/retest

It seems the previous build was aborted? No build logs are available..

**Comment by imdario** [comment]
Please, check this issue related to Kubernetes and Mergo: https://github.com/imdario/mergo/issues/33#issuecomment-357105051

I just released version 0.3.0, which introduces transformers and fixes several bugs.

**Comment by cblecker** [comment]
Closing out stale PR.

**Pull request closed without merging**